torch/_inductor/fx_passes/reinplace.py

```diff
@@ -421,6 +422,8 @@ def can_inplace(node, mutated_arg):
         if isinstance(mutated_arg, (list, tuple)):
             return all(can_inplace(node, arg) for arg in mutated_arg)
 
+        if mutated_arg is None:
```

Comment:
This feels slightly sketchy, when is this the case?

Reply:
Mmm, I can put this check somewhere less sketchy. The problem is that auto_functionalized can claim that it is mutating an input with type Optional[Tensor], and this check is needed for the case when the thing is None.

I can change it so that auto_functionalize doesn't think it's mutating the input if the type is None.

Reply:
Moved the check to an auto_functionalized-related location